src/argilla_server/validators/responses.py

```diff
@@ -61,37 +56,21 @@ def _validate_values(self, record: Record) -> None:
 
         for question in record.dataset.questions:
             if question_response := self._response_change.values.get(question.name):
-                if question.type == QuestionType.text:
-                    TextQuestionResponseValueValidator(question_response.value).validate()
-                elif question.type == QuestionType.label_selection:
-                    LabelSelectionQuestionResponseValueValidator(question_response.value).validate_for(
-                        question.parsed_settings
-                    )
-                elif question.type == QuestionType.multi_label_selection:
-                    MultiLabelSelectionQuestionResponseValueValidator(question_response.value).validate_for(
-                        question.parsed_settings
-                    )
-                elif question.type == QuestionType.rating:
-                    RatingQuestionResponseValueValidator(question_response.value).validate_for(question.parsed_settings)
-                elif question.type == QuestionType.ranking:
-                    RankingQuestionResponseValueValidator(question_response.value).validate_for(
-                        self._response_change.status, question.parsed_settings
-                    )
-                elif question.type == QuestionType.span:
-                    SpanQuestionResponseValueValidator(question_response.value).validate_for(
-                        question.parsed_settings, record
-                    )
-                else:
-                    raise ValueError(
-                        f"unknown question type f{question.type!r} for question with name f{question.name}"
-                    )
-
-
-class ResponseUpdateValidator(ResponseCreateValidator):
-    def __init__(self, response_change: ResponseUpdate):
-        self._response_change = response_change
+                ResponseValueValidator(question_response.value).validate_for(
+                    question.parsed_settings, record, self._response_change.status
+                )
 
 
-class ResponseUpsertValidator(ResponseCreateValidator):
-    def __init__(self, response_change: ResponseUpsert):
-        self._response_change = response_change
+class ResponseCreateValidator(ResponseValidator):
+    def __init__(self, response_create: ResponseCreate):
+        self._response_change = response_create
+
+
+class ResponseUpdateValidator(ResponseValidator):
+    def __init__(self, response_update: ResponseUpdate):
+        self._response_change = response_update
+
+
+class ResponseUpsertValidator(ResponseValidator):
+    def __init__(self, response_upsert: ResponseUpsert):
+        self._response_change = response_upsert
```

Comment:
With a generic repsonse validator you can define those as:
```suggestion
ResponseCreateValidator = ResponseValidator[ResponseCreate]
ResponseUpdateValidator = ResponseValidator[ResponseUpdate]
ResponseUpsertValidator = ResponseValidator[ResponseUpsert]
```